src/setupWorker/glossary.ts

```diff
@@ -181,10 +181,11 @@ export interface SerializedResponse<BodyType = any> {
   body: BodyType
 }
 
+export type PartialStartOptions = PartialDeep<StartOptions>
```

Comment:
This may be a good chance to review if we need a non-partial `StartOptions` anywhere in the code. 

I don't think that introducing this custom type is a good idea. Or, to be precise, I don't think having to do `PartialDeep<StartOptions>` everywhere is a good idea. I think we should mark `StartOptions` as optional on a per-property basis, utilizing `Partial` for options that equal objects (like `options.serviceWorker`). 

Please, would you be interested in looking into this? If you're strict on time, I can take it from here and let you know.

Reply:
I unfortunately do not have time at the moment to dig deeper in the codebase. I'd appreciate if you could take it from here and I'd love to continue watching the PR!

Reply:
No worries. I've added this suggestion. The only thing that remains is to export the `SetupServerApi`.